AsyncDisplayKit/ASCollectionView.h

```diff
@@ -342,18 +353,34 @@
  */
 - (BOOL)shouldBatchFetchForCollectionView:(ASCollectionView *)collectionView;
 
+@end
+
+@protocol ASCollectionViewDelegateFlowLayout <ASCollectionViewDelegate>
+
+@optional
+
 /**
  * Passthrough support to UICollectionViewDelegateFlowLayout sectionInset behavior.
  *
  * @param collectionView The sender.
  * @param collectionViewLayout The layout object requesting the information.
- * #param section The index number of the section whose insets are needed.
+ * @param section The index number of the section whose insets are needed.
  *
  * @discussion The same rules apply as the UICollectionView implementation, but this can also be used without a UICollectionViewFlowLayout.
  * https://developer.apple.com/library/ios/documentation/UIKit/Reference/UICollectionViewDelegateFlowLayout_protocol/index.html#//apple_ref/occ/intfm/UICollectionViewDelegateFlowLayout/collectionView:layout:insetForSectionAtIndex:
  *
  */
-- (UIEdgeInsets)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout insetForSectionAtIndex:(NSInteger)section;
+- (UIEdgeInsets)collectionView:(ASCollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout insetForSectionAtIndex:(NSInteger)section;
+
+/**
+ * Asks the delegate for the size of the header in the specified section.
+ */
+- (CGSize)collectionView:(ASCollectionView *)collectionView layout:(UICollectionViewLayout *)collectionViewLayout referenceSizeForHeaderInSection:(NSInteger)section;
```

Comment:
I'm still skeptical about this.  There could be a small performance win from not having to create nodes, but I think the downside of the API surface area is greater — particularly because this is crossing concepts that developers may already understand (e.g. it is not normally on UICollectionView, but rather UICollectionViewFlowLayout).

I could fix the performance cost entirely by providing a +staticSize or similar method on ASCellNode, which, if it returns a non-CGSizeZero, would be used instead of allocating the node at that time.  The only bummer is that these methods will be tough to remove later.

I guess in the interest of time, the best plan is to just make a breaking API change in the future when they're removed.  

@levi - Is it already the case that if they are not implemented, the node IS instantiated and consulted for size?  Or does it not give any space for the header / footer in that case?